cram/cram_io.c

```diff
@@ -5564,7 +5571,7 @@ int cram_close(cram_fd *fd) {
     if (fd->mode == 'w') {
         /* Write EOF block */
         if (0 != cram_write_eof_block(fd))
-            return -1;
+            ret = -1;
```

Comment:
It might be an idea to avoid writing the eof block if an error has already been detected.